x/datapool/keeper/pool.go

```diff
@@ -553,3 +544,129 @@ func (k Keeper) increaseNumIssuedNFT(ctx sdk.Context, pool *types.Pool) {
 
 	k.SetPool(ctx, pool)
 }
+
+func (k Keeper) RedeemDataPass(ctx sdk.Context, redeemNFT types.MsgRedeemDataPass) (*types.DataPassRedeemReceipt, error) {
```

Comment:
What if the same person plays Redeem multiple times?

Reply:
When redeemer redeems data pass, the data pass(NFT) is transferred to module account, so replay attack can't happen. Did you mean this kind of replay attack, right?

Reply:
Oh sorry i misunderstood
I thought this function was looking up RedeemDataPass.
You can ignore this comment.

Reply:
> When redeemer redeems data pass, the data pass(NFT) is transferred to module account, so replay attack can't happen. Did you mean this kind of replay attack, right?
As @hansol-medi said, replay attacks are not a problem.